pkg/kubelet/kuberuntime/helpers.go

```diff
@@ -217,18 +217,21 @@ func toKubeRuntimeStatus(status *runtimeapi.RuntimeStatus, handlers []*runtimeap
 			Message: c.Message,
 		})
 	}
-	retHandlers := make(map[string]kubecontainer.RuntimeHandler)
+	rtClasses := []v1.RuntimeClass{}
 	for _, h := range handlers {
-		supportsUserns := false
-		if h.Features != nil {
-			supportsUserns = h.Features.UserNamespaces
+		rtClass := v1.RuntimeClass{
+			Name:     h.Name,
+			Features: &v1.RuntimeClassFeatures{},
 		}
-		retHandlers[h.Name] = kubecontainer.RuntimeHandler{
-			Name:                   h.Name,
-			SupportsUserNamespaces: supportsUserns,
+		// The runtime features are exposed regardless to the values of the Kubernetes feature gates.
+		// https://github.com/kubernetes/kubernetes/pull/123216#discussion_r1484762897
```

Comment:
cc @mrunalp

Reply:
I am testing with this PR on a local cluster without the feature gate enabled and I see that:
1. If I use this example
```
apiVersion: v1
kind: Pod
metadata:
  name: rro
spec:
  volumes:
    - name: mnt
      hostPath:
        # tmpfs is mounted on /mnt/tmpfs
        path: /mnt
  containers:
    - name: busybox
      image: busybox
      args: ["sleep", "infinity"]
      volumeMounts:
        # /mnt-rro/tmpfs is not writable
        - name: mnt
          mountPath: /mnt-rro
          readOnly: true
          mountPropagation: None
          recursiveReadOnly: IfPossible
```

then the pod goes to running with the status of rro as Disabled. (See my other comment on the code).

2. If the Feature Gate isn't enabled for User Namespaces for example, then we do something like this:
https://github.com/kubernetes/kubernetes/blob/master/pkg/api/pod/util.go#L552C1-L559C3

```
	// If the feature is disabled and not in use, drop the hostUsers field.
	if !utilfeature.DefaultFeatureGate.Enabled(features.UserNamespacesSupport) && !hostUsersInUse(oldPodSpec) {
		// Drop the field in podSpec only if SecurityContext is not nil.
		// If it is nil, there is no need to set hostUsers=nil (it will be nil too).
		if podSpec.SecurityContext != nil {
			podSpec.SecurityContext.HostUsers = nil
		}
	}
```

Reply:
I will defer to API review (@thockin ) for this.

Reply:
Updated `dropDisabledFields`